### DISCUSSION TITLE
Define special function like `Array.sort`

### DISCUSSION THREAD

**Discussion by HKhademian** [vlang/v#15211] [General]
When I implement a Tree, I use a `fn` as input to my function.
Then when I need to call that function, I must provide a full fn, like :
```vlang
n := &Node<string>{...}
n.bfs(fn (n string) {
	println(n)
})
```

But I saw some builtin functions like [Array.sort](https://github.com/vlang/v/blob/master/doc/docs.md#sorting-arrays) or [Array.filter or Array.map](https://github.com/vlang/v/blob/master/doc/docs.md#array-method-chaining) uses a **special variable** like `a`, `b`, `it`.

How can I define my
```vlang
fn (n &Node) bfs(processor fn ???)
{ ... }

n.bfs( it.print() )
```
to use these syntax sugar?

**Comment by larpon** [comment]
You currently can't, since `it`, `a` and `b` is compiler magic

**Comment by HKhademian** [comment]
Can we make it a feature request?

**Reply by larpon** [reply]
Go ahead 👍🏻 Everyone is free to open feature requests via GitHub issues

**Reply by larpon** [reply]
https://github.com/vlang/v/issues/new/choose
